### PR TITLE
Functional tests - Refactoring products tests

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Refactoring products page and products tests 
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | Sanity tests should pass

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18112)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

File: tests/puppeteer/pages/BO/BObasePage.js [hunk 1]
```diff
@@ -275,11 +277,30 @@ module.exports = class BOBasePage extends CommonPage {
     return this.getAttributeContent(this.helpDocumentURL, 'data');
   }
 
+  /**
+   * Check if Submenu is visible
+   * @param parentSelector
+   * @param linkSelector
+   * @return {Promise<boolean|true>}
```
Comment (nesrineabdmouleh) [inline tests/puppeteer/pages/BO/BObasePage.js]:
```suggestion
   * @return {Promise<boolean>}
```

File: tests/puppeteer/pages/BO/catalog/products/add.js [hunk 2]
```diff
@@ -308,11 +311,21 @@ module.exports = class AddProduct extends BOBasePage {
     return this.getAttributeContent(this.friendlyUrlInput, 'value');
   }
 
+  /**
+   * Add specific prices
+   * @param specificPriceData
+   * @return {Promise<void>}
```
Comment (nesrineabdmouleh) [inline tests/puppeteer/pages/BO/catalog/products/add.js]:
```suggestion
   * @return {Promise<string>}
```

File: tests/puppeteer/pages/BO/BObasePage.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/BObasePage.js]: done thanks

File: tests/puppeteer/pages/BO/catalog/products/add.js (hunk 2 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/catalog/products/add.js]: done thanks

File: tests/puppeteer/pages/FO/product.js [hunk 3]
```diff
@@ -23,15 +23,14 @@ module.exports = class Product extends FOBasePage {
...
    */
-  async checkProduct(productData) {
+  async getProductInformation(productData) {
     return {
-      name: await this.checkTextValue(this.productName, productData.name),
-      price: await this.checkAttributeValue(this.productPrice, 'content', productData.price),
-      quantity_wanted: await this.checkAttributeValue(this.productQuantity, 'value', productData.quantity_wanted),
-      description: await this.checkTextValue(this.productDescription, productData.description, 'contain'),
+      name: await this.getTextContent(this.productName, productData.name),
+      price: parseFloat(await this.getAttributeContent(this.productPrice, 'content', productData.price)),
```
Comment (nesrineabdmouleh) [inline tests/puppeteer/pages/FO/product.js]:
```suggestion
      price: parseFloat(await this.getAttributeContent(this.productPrice, 'content')),
```

Comment (boubkerbribri) [reply tests/puppeteer/pages/FO/product.js]: done thanks

File: tests/puppeteer/pages/commonPage.js [hunk 4]
```diff
@@ -154,14 +154,12 @@ module.exports = class CommonPage {
   /**
    * Check attribute value
    * @param selector, element to check
-   * @param attribute, attribute to test
+   * @param attribute, attribute to get
    * @param textToCheckWith, text to check with
    * @return promise, throw an error if element does not exist or attribute value is not correct
    */
   async checkAttributeValue(selector, attribute, textToCheckWith) {
```
Comment (nesrineabdmouleh) [inline tests/puppeteer/pages/commonPage.js]: Maybe this function is not useful because we are checking the values in the tests.

Comment (boubkerbribri) [reply tests/puppeteer/pages/commonPage.js]: Nice catch, deleted :+1:

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**